code/datums/storage/storage.dm

```diff
@@ -0,0 +1,1029 @@
+		INVOKE_ASYNC(src, PROC_REF(attempt_draw_object), user)
+
+///Called when you alt + right click on parent
+///Opens the inventory of our storage
+/datum/storage/proc/on_alt_right_click(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(parent.Adjacent(user))
+		open(user)
+
+///Called when you ctrl + left click on parent
+///Attempts to draw an object from out storage, but it draw from the left side instead of the right
+/datum/storage/proc/on_ctrl_click(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(parent.Adjacent(user))
+		INVOKE_ASYNC(src, PROC_REF(attempt_draw_object), user, TRUE)
+
+/datum/storage/proc/on_attack_ghost(datum/source, mob/user)
+	SIGNAL_HANDLER
+	open(user)
+
+///Signal handler for when you click drag parent to something (usually ourselves or an inventory slot)
+/datum/storage/proc/on_mousedrop_onto(datum/source, obj/over_object as obj, mob/user)
+	SIGNAL_HANDLER
+	if(!ishuman(user))
+		return COMPONENT_NO_MOUSEDROP
+
+	if(user.lying_angle)
+		return COMPONENT_NO_MOUSEDROP
+
+	if(over_object == user && parent.Adjacent(user)) // this must come before the screen objects only block
+		open(user)
+		return COMPONENT_NO_MOUSEDROP
+
+	if(!istype(over_object, /atom/movable/screen))
+		return //Don't cancel mousedrop
+
+	if(HAS_TRAIT(src, TRAIT_NODROP))
+		return COMPONENT_NO_MOUSEDROP
+
+	//Makes sure that the storage is equipped, so that we can't drag it into our hand from miles away.
+	if(parent.loc != user && parent.loc.loc != user) //loc.loc handles edge case of storage attached to an item attached to another item (modules/boots)
+		return COMPONENT_NO_MOUSEDROP
+
+	if(!user.restrained() && !user.stat)
+		switch(over_object.name)
+			if("r_hand")
+				user.temporarilyRemoveItemFromInventory(source)
+				if(!user.put_in_r_hand(source))
+					user.dropItemToGround(source)
+				return COMPONENT_NO_MOUSEDROP
+			if("l_hand")
+				user.temporarilyRemoveItemFromInventory(source)
+				if(!user.put_in_l_hand(source))
+					user.dropItemToGround(source)
+				return COMPONENT_NO_MOUSEDROP
+
+/datum/storage/verb/toggle_gathering_mode()
+	set name = "Switch Gathering Method"
+	set category = "Object"
+
+	collection_mode = !collection_mode
+	switch (collection_mode)
+		if(1)
+			to_chat(usr, "[parent.name] now picks up all items in a tile at once.")
+		if(0)
+			to_chat(usr, "[parent.name] now picks up one item at a time.")
+
+/datum/storage/verb/toggle_draw_mode()
+	set name = "Switch Storage Drawing Method"
+	set category = "Object"
+	draw_mode = !draw_mode
+	if(draw_mode)
+		to_chat(usr, "Clicking [parent.name] with an empty hand now puts the last stored item in your hand.")
+	else
+		to_chat(usr, "Clicking [parent.name] with an empty hand now opens the pouch storage menu.")
+
+/**
+ * Gets the inventory of a storage
+ * if recursive = TRUE, this will also get the inventories of things within the inventory
+ */
+/datum/storage/proc/return_inv(recursive = TRUE)
+	var/list/inventory = list()
+	inventory += parent.contents
```

Comment:
cant you do inventory = parent.contents.copy()

Reply:
No, because of the way BYOND handles contents being not a real list.
My main incentive to making this PR was because I was upset when I found out that you are also unable to call .Swap() on .contents for the same reason.